Framework/Core/src/ExternalFairMQDeviceProxy.cxx

```diff
@@ -152,11 +161,21 @@ DataProcessorSpec specifyExternalFairMQDeviceProxy(char const* name,
   // messages into DPL messages.
   spec.algorithm = AlgorithmSpec{[converter, channel = spec.name](InitContext& ctx) {
     auto device = ctx.services().get<RawDeviceService>().device();
+    auto outputs = ctx.services().get<RawDeviceService>().spec().outputs;
     assert(device);
 
     // Converter should pump messages
-    auto handler = [device, converter](FairMQParts& inputs, int idx) {
-      converter(*device, inputs, idx);
+    auto handler = [device, converter, outputs = std::move(outputs)](FairMQParts& inputs, int) {
+      auto channelRetreiver = [outputs = std::move(outputs)](OutputSpec const& os) {
+        for (auto& output : outputs) {
+          auto query = DataSpecUtils::asConcreteDataMatcher(os);
```

Comment:
This throws when OutputSpec has `ConcreteDataTypeMatcher`, but not `ConcreteDataMatcher`, which breaks access to readout data in QC. Using `ConcreteDataTypeMatcher` allows me to publish data with any `subSpec`, so I can map `DataBlockHeaderBase.linkID` into `subSpec`.
Could you please support it this case as well?

Reply:
hanks for looking at this, yes, will do

Reply:
@knopers8 should be fixed now